### PR TITLE
cmd/tailscale/cli: [serve] fix MinGW path conversion

### LINKED ISSUE #7963: MinGW shell path conversion mangles Serve mount-points
See the Reddit thread:
https://www.reddit.com/r/Tailscale/comments/12y1ttd/im_trying_to_use_funnel_to_publicly_access_a/

It looks to be a [feature, not a bug](https://web.archive.org/web/20120222092002/http://www.mingw.org/wiki/Posix_path_conversion), as well as a [known issue](https://github.com/git-for-windows/build-extra/blob/main/ReleaseNotes.md#known-issues).

Temporarily setting the env var `MSYS_NO_PATHCONV=1` will prevent the conversion. However, exporting it can cause issues.
The other workaround is to double the first slash `//`, which avoids the "POSIX-to-Windows" path conversion.

We can detect and fix this at runtime as well.

### PR DESCRIPTION
**Pull request by shayne**
Fixes #7963

Signed-off-by: Shayne Sweeney <shayne@tailscale.com>

### CODE REVIEW

**Review by bradfitz — APPROVED** [review]
Pull this out into a function probably? I bet it won't be the last time we encountered this.

File: cmd/tailscale/cli/serve.go [hunk 1]
```diff
@@ -412,6 +412,15 @@ func cleanMountPoint(mount string) (string, error) {
 	if mount == "" {
 		return "", errors.New("mount point cannot be empty")
 	}
+	// MinGW(ish) (Windows) shells perform POSIX-to-Windows path conversion
+	// converting the leading "/" of any shell arg to the EXEPATH, which mangles
+	// the mount point. Strip the EXEPATH prefix if it exists. #7963
+	//
+	// "/C:/Program Files/Git/foo" -> "/foo"
+	if _, ok := os.LookupEnv("MSYSTEM"); ok {
```
Comment (bradfitz) [inline cmd/tailscale/cli/serve.go:420]: And make the function only do anything on windows?

File: cmd/tailscale/cli/serve.go [hunk 2]
```diff
@@ -412,6 +413,17 @@ func cleanMountPoint(mount string) (string, error) {
 	if mount == "" {
 		return "", errors.New("mount point cannot be empty")
 	}
+	// MinGW(ish) (Windows) shells perform POSIX-to-Windows path conversion
+	// converting the leading "/" of any shell arg to the EXEPATH, which mangles
+	// the mount point. Strip the EXEPATH prefix if it exists. #7963
+	//
+	// "/C:/Program Files/Git/foo" -> "/foo"
+	if runtime.GOOS != "windows" {
```
Comment (bradfitz) [inline cmd/tailscale/cli/serve.go:421]: backwards?

Comment (shayne) [reply cmd/tailscale/cli/serve.go:421]: YUP

Comment (shayne) [reply cmd/tailscale/cli/serve.go]: Updated.

**Review by bradfitz — APPROVED** [review]

**Pull request merged**